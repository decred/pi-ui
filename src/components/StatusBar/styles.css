.statusWrapper {
  display: flex;
  margin-top: 1rem;
  position: relative;
}

.statusInfo {
  display: flex;
  justify-content: space-between;
}

.infoTotal {
  color: var(--color-gray);
}

.legendLine {
  display: flex;
  align-items: center;
}

.legendLine:not(:last-child) {
  margin-right: 3rem;
}

.legend {
  display: flex;
}

.legendColor {
  width: 0.6rem;
  height: 0.6rem;
  border-radius: 50%;
  margin-right: 0.5rem;
}

.legendLabel {
  font-weight: var(--font-weight-extra-bold);
  margin-right: 0.5rem;
}

.legendAmount {
  margin-right: 0.5rem;
}

.statusOption {
  height: 8px;
}

.firstStatusOption {
  border-radius: 10rem 0 0 10rem;
}

.lastStatusOption {
  border-radius: 0 10rem 10rem 0;
}

.marker {
  height: 20px;
  width: 2px;
  background-color: var(--color-gray-light);
  position: absolute;
  transform: translate(0, -25%);
}
