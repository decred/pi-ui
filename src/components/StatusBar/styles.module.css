.statusBar {
  width: 100%;
}

.statusWrapper {
  display: flex;
  margin-top: 0.5rem;
  position: relative;
  border: 1px solid var(--color-gray-light);
  border-radius: 10rem;
  height: 10px;
}

.statusBar.balance .statusWrapper > div {
  flex: 1;
  display: flex;
  flex-direction: row;
}

.statusBar.balance .statusWrapper > div:first-of-type {
  justify-content: flex-end;
}

.statusWrapperDark {
  border: 1px solid var(--icon-color);
}

.statusInfo {
  display: flex;
  flex-wrap: wrap-reverse;
  justify-content: space-between;
  align-items: center;
}

.infoCurrent {
  font-size: var(--font-size-small);
  line-height: var(--spacing-1);
  font-weight: var(--font-weight-regular);
}

.infoTotal {
  composes: infoCurrent;
  color: var(--color-gray);
}

.legendLine {
  display: flex;
  align-items: center;
  color: var(--tab-text-active-color);
}

.legendLine:not(:last-child) {
  margin-right: 3rem;
}

.statusBar.balance .legendLine:last-of-type {
  position: absolute;
  left: 50%;
  padding-left: 35px;
}

.legend {
  composes: infoCurrent;
  display: flex;
  margin-right: 2rem;
}

.legendColor {
  width: 0.6rem;
  height: 0.6rem;
  border-radius: 50%;
  margin-right: 0.5rem;
}

.legendLabel {
  margin-right: 0.5rem;
  text-transform: capitalize;
  font-weight: var(--font-weight-semi-bold);
}

.legendAmount {
  margin-right: 0.5rem;
}

.statusOption {
  height: 8px;
}

.statusOption:first-of-type {
  border-radius: 10rem 0 0 10rem;
}

.statusOption:last-of-type {
  border-radius: 0 10rem 10rem 0;
}

.statusBar.balance .statusWrapper > div:first-of-type .statusOption {
  border-radius: 10rem 0 0 10rem;
}

.markerWrapper {
  position: absolute;
  transform: translate(-50%, -25%);
}

.marker {
  height: 20px;
  width: 2px;
  cursor: pointer;
  background-color: var(--color-gray-light);
}

.marker.markerDark {
  background-color: var(--color-primary-dark);
}

.markerTooltip {
  padding: 0 0.3rem;
  cursor: pointer;
  color: var(--color-green);
  font-size: var(--font-size-small);
}

@media screen and (max-width: 768px) {
  .legendLine:not(:last-child) {
    margin-right: 1rem;
  }
}
@custom-media --xs-viewport (max-width: 560px);


@media screen and (--xs-viewport) {
  .statusInfo {
    flex-direction: column-reverse;
    align-items: flex-end;
  }
}
