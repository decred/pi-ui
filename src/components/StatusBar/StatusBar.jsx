import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";

const StatusBar = ({ status, markerPosition, max }) => {
  const currentTotal = status.reduce((acc, cur) => acc + cur.amount, 0);
  const statusWithPercentages = status.map((item) => ({
    ...item,
    percentage: (item.amount / currentTotal) * 100
  }));
  return (
    <div>
      <div className={styles.statusInfo}>
        <div className={styles.legend}>
          {statusWithPercentages.map((st, i) => (
            <div className={styles.legendLine}>
              <div
                style={{ backgroundColor: st.color }}
                className={styles.legendColor}
              />
              <span className={styles.legendLabel}>{st.label}:</span>
              <span className={styles.legendAmount}>{st.amount}</span>
              <span>({st.percentage}%)</span>
            </div>
          ))}
        </div>
        <span>
          <span>{currentTotal}</span>
          <span className={styles.infoTotal}>/{max} votes</span>
        </span>
      </div>
      <div className={styles.statusWrapper}>
        {statusWithPercentages.map((st) => (
          <div
            style={{
              height: "8px",
              width: `${st.percentage}%`,
              backgroundColor: st.color
            }}
          />
        ))}
        <div
          className={styles.marker}
          style={{
            left: markerPosition
          }}
        />
      </div>
    </div>
  );
};

StatusBar.propTypes = {
  status: PropTypes.array.isRequired,
  markerPosition: PropTypes.string,
  max: PropTypes.number
};

StatusBar.defaultProps = {
  markerPosition: "0%",
  max: 0
};

export default StatusBar;
