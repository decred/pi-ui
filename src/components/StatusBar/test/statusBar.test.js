import React from "react";
import StatusBar from "../StatusBar";
import { create } from "react-test-renderer";
import {
  defaultLightTheme,
  ThemeProvider,
  DEFAULT_LIGHT_THEME_NAME,
} from "../../../theme";

describe("StatusBar component", () => {
  test("Matches the snapshot", () => {
    const statusBar = create(
      <ThemeProvider
        themes={{ [DEFAULT_LIGHT_THEME_NAME]: defaultLightTheme }}
        defaultThemeName={DEFAULT_LIGHT_THEME_NAME}
      >
        <StatusBar
          status={[
            {
              label: "Yes",
              amount: 300,
              color: "green",
            },
            {
              label: "No",
              amount: 200,
              color: "orange",
            },
          ]}
        />
      </ThemeProvider>
    );
    expect(statusBar.toJSON()).toMatchSnapshot();
  });
});
