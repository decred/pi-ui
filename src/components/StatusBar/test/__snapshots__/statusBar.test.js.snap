// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar component Matches the snapshot 1`] = `
<div
  className="statusBar"
>
  <div
    className="statusInfo"
  >
    <div
      className="legend"
    >
      <div
        className="legendLine"
      >
        <div
          className="legendColor"
          style={
            Object {
              "backgroundColor": "green",
            }
          }
        />
         
        <span
          className="legendLabel"
        >
          Yes
          :
        </span>
        <span
          className="legendAmount"
        >
          300
        </span>
        <span>
          (
          60.0
          %)
        </span>
      </div>
      <div
        className="legendLine"
      >
        <div
          className="legendColor"
          style={
            Object {
              "backgroundColor": "orange",
            }
          }
        />
         
        <span
          className="legendLabel"
        >
          No
          :
        </span>
        <span
          className="legendAmount"
        >
          200
        </span>
        <span>
          (
          40.0
          %)
        </span>
      </div>
    </div>
    <span>
      <span
        className="infoCurrent"
      >
        500
      </span>
      <span
        className="infoTotal"
      >
        /
        0
         votes
      </span>
    </span>
  </div>
  <div
    className="statusWrapper"
  >
    <div
      className="statusOption"
      style={
        Object {
          "backgroundColor": "green",
          "width": "60%",
        }
      }
    />
    <div
      className="markerWrapper"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <div
        className="tooltip markerTooltip"
        onClick={[Function]}
      >
        <div
          className="tooltipContent top"
        >
          0%
        </div>
        <div
          className="marker"
        />
      </div>
    </div>
    <div
      className="statusOption"
      style={
        Object {
          "backgroundColor": "orange",
          "width": "40%",
        }
      }
    />
  </div>
</div>
`;
